### PR TITLE
Content for configuring Azure credential source.

### LINKED ISSUE #639: Need documentation for CredentialSource in Azure provisioner configuration
Original PR: https://github.com/dotnet/aspire/pull/3083

This feature is current exposed in Preview 5 with no documentation.

See its use here to fix an issue https://github.com/dotnet/aspire/issues/3544

Ideally, we should include an example scenario that encompasses the configuration of some components (i.e. also at runtime, not only deployment) to use the same non-default credential, as this would be the most likely use-case.

i.e. 

AppHost:

```json
{
    "Azure": {
        "CredentialSource": "AzureCli"
    }
}
```

ProjectA

```csharp
builder.AddAzureEventHubs("myeh", settings => settings.Credential = new AzureCliCredential());
```

---
[Associated WorkItem - 244313](https://dev.azure.com/msft-skilling/Content/_workitems/edit/244313)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Add details to all Azure resource-related content, with provisioning and credential source config.

Fixes #639

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/azureai/azureai-openai-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/azureai/azureai-openai-component.md) | [.NET Aspire Azure AI OpenAI component](https://review.learn.microsoft.com/en-us/dotnet/aspire/azureai/azureai-openai-component?branch=pr-en-us-693) |
| [docs/azureai/azureai-search-document-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/azureai/azureai-search-document-component.md) | [NET Aspire Azure AI Search Documents component](https://review.learn.microsoft.com/en-us/dotnet/aspire/azureai/azureai-search-document-component?branch=pr-en-us-693) |
| [docs/database/azure-cosmos-db-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/database/azure-cosmos-db-component.md) | [.NET Aspire Azure Cosmos DB component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/azure-cosmos-db-component?branch=pr-en-us-693) |
| [docs/database/azure-cosmos-db-entity-framework-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/database/azure-cosmos-db-entity-framework-component.md) | [.NET Aspire Microsoft Entity Framework Core Cosmos DB component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/azure-cosmos-db-entity-framework-component?branch=pr-en-us-693) |
| [docs/messaging/azure-event-hubs-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/messaging/azure-event-hubs-component.md) | [docs/messaging/azure-event-hubs-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/azure-event-hubs-component?branch=pr-en-us-693) |
| [docs/messaging/azure-service-bus-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/messaging/azure-service-bus-component.md) | [.NET Aspire Azure Service Bus component](https://review.learn.microsoft.com/en-us/dotnet/aspire/messaging/azure-service-bus-component?branch=pr-en-us-693) |
| [docs/real-time/azure-signalr-scenario.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/real-time/azure-signalr-scenario.md) | [.NET Aspire support for Azure SignalR Service](https://review.learn.microsoft.com/en-us/dotnet/aspire/real-time/azure-signalr-scenario?branch=pr-en-us-693) |
| [docs/security/azure-security-key-vault-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/security/azure-security-key-vault-component.md) | [.NET Aspire Azure Key Vault component](https://review.learn.microsoft.com/en-us/dotnet/aspire/security/azure-security-key-vault-component?branch=pr-en-us-693) |
| [docs/storage/azure-storage-blobs-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/storage/azure-storage-blobs-component.md) | [.NET Aspire Azure Blob Storage component](https://review.learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-blobs-component?branch=pr-en-us-693) |
| [docs/storage/azure-storage-queues-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/storage/azure-storage-queues-component.md) | [docs/storage/azure-storage-queues-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-queues-component?branch=pr-en-us-693) |
| [docs/storage/azure-storage-tables-component.md](https://github.com/dotnet/docs-aspire/blob/7d2990495022881f5a090fae63511e3ceb372ed8/docs/storage/azure-storage-tables-component.md) | [.NET Aspire Azure Data Tables component](https://review.learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-tables-component?branch=pr-en-us-693) |

</details>

### CODE REVIEW

File: docs/includes/azure-provisioning.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## Azure provisioning credential store
+
+The .NET Aspire app host uses a credential store for Azure resource authentication and authorization. Depending on your subscription, the correct credential store may be needed for multi-tenant provisioning scenarios.
+
+With the [Aspire.Hosting.Azure](https://nuget.org/packages/Aspire.Hosting.Azure) NuGet package installed, and if your app host depends on Azure resources, the default Azure credential store relies on the <xref:Azure.Identity.DefaultAzureCredential>. To change this behavior, and you can set the credential store value in the _appsettings.json_ file, as shown in the following example:
```
Comment (tdykstra) [inline docs/includes/azure-provisioning.md:5]:
```suggestion
With the [Aspire.Hosting.Azure](https://nuget.org/packages/Aspire.Hosting.Azure) NuGet package installed, and if your app host depends on Azure resources, the default Azure credential store relies on the <xref:Azure.Identity.DefaultAzureCredential>. To change this behavior, you can set the credential store value in the _appsettings.json_ file, as shown in the following example:
```

**Review by tdykstra — APPROVED** [review]
LGTM, one issue found.

File: docs/real-time/snippets/signalr/SignalR.AppHost/Program.cs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 ﻿var builder = DistributedApplication.CreateBuilder(args);
 
+builder.AddAzureProvisioning();
```
Comment (davidfowl) [inline docs/real-time/snippets/signalr/SignalR.AppHost/Program.cs:3]: This isn't needed.

File: docs/includes/azure-provisioning.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+## Azure provisioning credential store
```
Comment (davidfowl) [inline docs/includes/azure-provisioning.md:1]: This only applies to local provisioning.

**Comment by davidfowl** [comment]
The azure credentials only works for local provisioning

**Pull request merged**